### PR TITLE
React Router Lesson: Add small blurp/section on why use router #24210

### LINKED ISSUE #24210: React Router Lesson: Add small blurp/section on why use router
Complete the following REQUIRED checkboxes:
-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this issue follows the `location for request: brief description of request` format, e.g. `NodeJS course: Add lessons on XYZ`

The following checkbox is OPTIONAL:

-   [ ] I would like to be assigned this issue to work on it

<hr>

**1. Description of the Feature Request:**

**2. Acceptance Criteria:**

**3. Additional Information:**

When I was completing this section I was wondering why even use router as opposed to just using links. I think highlighting the fact that router prevents a screen flash/refresh at the beginning of the lesson could help people get excited about learning router, while at the same time helping them understand its benefit. Maybe the following link could be helpful as well. https://medium.com/@marcellamaki/a-brief-overview-of-react-router-and-client-side-routing-70eb420e8cde

### PR DESCRIPTION
**Pull request by Lakshmikant-2001**
Complete the following REQUIRED checkboxes:

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`

Complete the following checkboxes ONLY IF they are applicable to your PR. You can complete them later if they are not currently applicable:
-   [x] I have previewed all lesson files included in this PR with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure the Markdown content is formatted correctly
-   [x] I have ensured all lesson files included in this PR follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

<hr>

**1. Because:**

Closes #24210 

**2. This PR:**
Satisfies the Acceptance Criteria, 
- [x] Add a section called "Client-side Routing" right after Learning Outcomes
- [x] In the new section, discuss client-side routing and why it's useful. Then, briefly touch upon how React Router is handy for routing in React applications.
- [x] Rename the "React Router" section to "How To Use React Router"

### CODE REVIEW

File: javascript/react_js/router.md [hunk 1]
```diff
@@ -5,9 +5,16 @@ In this lecture we are going to learn a little bit about routing in React. Up un
 ### Learning Outcomes
 By the end of this lesson, you should be able to:
 
+- Understand what client-side routing is
 - Explain how `react-router-dom` works
 
-### React Router
+### Client-side Routing
```
Comment (01zulfi) [inline javascript/react_js/router.md:11]: As per our [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md) we want our headings to be [Wikipedia Style Title Case](https://titlecaseconverter.com/rules/#WP):
```suggestion
### Client-Side Routing
```

File: javascript/react_js/router.md [hunk 2]
```diff
@@ -5,9 +5,16 @@ In this lecture we are going to learn a little bit about routing in React. Up un
...
+
+<span id="client-side-routing">Client side routing is internal handling of routes inside the JS file that is rendered to the client(front-end). Client Side Routing helps in building Single-page applications(SPAs) without refreshing as the user navigates. For example when a user clicks a navbar element, the URL changes and the view of the page is modified accordingly, within the client.</span>
+ 
+React router is the standard routing library for React Applications. By using React routers, we can specify which component can be rendered based on the route. From version 4, react router uses dynamic routes (routing that takes place as your app is rendering).
```
Comment (01zulfi) [inline javascript/react_js/router.md:15]: I think we could add a reference link for dynamic routes:
```suggestion
React router is the standard routing library for React Applications. By using React routers, we can specify which component can be rendered based on the route. From version 4, react router uses [dynamic routes](https://v5.reactrouter.com/web/guides/philosophy/dynamic-routing) (routing that takes place as your app is rendering).
```

File: javascript/react_js/router.md [hunk 3]
```diff
@@ -5,9 +5,16 @@ In this lecture we are going to learn a little bit about routing in React. Up un
...
+
+### How to use React Router
```
Comment (01zulfi) [inline javascript/react_js/router.md:17]: See suggestion about lesson titles above.

```suggestion
### How To Use React Router
```

**Review by 01zulfi — CHANGES_REQUESTED** [review]
Great work, especially with including learning outcomes and knowledge checks :fire: 

I've made some minor requested changes below.

**Comment by Lakshmikant-2001** [comment]
@01zulfi Resolved all the comments.

**Review by 01zulfi — APPROVED** [review]

**Pull request merged**